### PR TITLE
Scripting functions required for SCPUI conversion of Red Alert

### PR DESCRIPTION
**Pull request by MjnMixael**
Should be everything required. Tested both committing to a mission and replaying the previous mission using these functions and it seems to work as expected.

### CODE REVIEW

**Review by wookieejedi — APPROVED** [review]
Particularly straight forward one. Looks good to me, and good to hear tests worked.

File: code/scripting/api/objs/redalert.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", current->stages[0].text.c_str());
+}
+
+ADE_VIRTVAR(AudioFilename, l_RedAlertStage, nullptr, "The audio file of the stage", "red_alert_stage", "The  audio file")
```
Comment (Goober5000) [inline code/scripting/api/objs/redalert.cpp:23]: "The  audio file" has an extra space

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Due to the way FSO handles briefings, this will need a slight redesign.

File: code/scripting/api/libs/ui.cpp [hunk 2]
```diff
@@ -491,5 +493,38 @@ ADE_FUNC(runBriefingStageHook, l_UserInterface_CmdBrief, "number oldStage, numbe
...
+	l_UserInterface);
+
+ ADE_FUNC(getRedAlert,
+	l_UserInterface_RedAlert,
+	nullptr,
+	"Get the red alert brief.",
+	"red_alert_stage",
+	"The loop brief data")
+{
+	return ade_set_args(L, "o", l_RedAlertStage.Set(Briefings[0]));
```
[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/scripting/api/libs/ui.cpp]: Hang on.. setting this as I think you intend causes a CTD when this method is called.

[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/scripting/api/libs/ui.cpp]: I think you mean `Briefings[0]`?

[the remark this answers is not in the document]
Comment (Goober5000) [reply code/scripting/api/libs/ui.cpp]: Yes, sorry, `Briefings` not `Briefing`.

[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/scripting/api/libs/ui.cpp]: In that case done

**Comment by MjnMixael** [comment]
Feedback addressed

**Review by Goober5000 — COMMENTED** [review]
That `Briefing` discussion got me thinking.  Does SCPUI call any of the built-in ui functions, such as `red_alert_do_frame()`, or is the entire UI handled by the script?  Many of these use the `Briefing` pointer.  Or in the case of standard briefings, there are some functions that use it that aren't even part of the UI, such as `brief_compact_stages()`.

The `Briefing` pointer is initialized to null but is set to the briefing in use once the UI screen is initialized.  For example, in in `brief_init()`:
```
	// get a pointer to the appropriate briefing structure
	if(MULTI_TEAM){
		Briefing = &Briefings[Net_player->p_info.team];
	} else {
		Briefing = &Briefings[0];			
	}
```
And in `red_alert_init()`:
```
	if ( !Briefing ) {
		Briefing = &Briefings[0];			
	}
```
In summary, if your SCPUI script calls any FSO functions that use the `Briefing` pointer, then it will need to make sure the pointer is initialized when the briefing is.

**Comment by MjnMixael** [comment]
None of this calls built-in functions (yet) and for data retrieval I try to avoid it.. though the Tech Room PR had to get a little fancier to initialize some of the data. Everything other than generating data is handled by the scripts before the game_state really gets through init.

**Review by Goober5000 — APPROVED** [review]
Ok.  In that case, everything looks good.
